### PR TITLE
Add `for of` support for iterator

### PR DESCRIPTION
**Pull request by tz70s**
Reference issue at #64

### CODE REVIEW

**Comment by mcollina** [comment]
Good work! Can you update the README as well? We should document this into the API section.

**Comment by tz70s** [comment]
@mcollina how does this look like?

**Comment by tz70s** [comment]
Cool! I add `for of` construct on the main bloomrun object for looped over all available patterns. But not sure the documentation is correct?

File: bloomrun.js [hunk 1]
```diff
@@ -152,4 +152,14 @@ Bloomrun.prototype.iterator = function (obj, opts) {
...
 }
 
+Bloomrun.prototype[Symbol.iterator] = function () {
+  var iterator = this.iterator()
+  return {
+    next: function () {
+      var result = iterator.next()
+      return (result !== null) ? { value: result, done: false } : { done: true }
+    }
+  }
```
Comment (mcollina) [inline bloomrun.js]: you can just call iterator[Symbol.iterator]() here and reuse the code in the Iterator itself.

File: README.md [hunk 2]
```diff
@@ -138,6 +138,17 @@ form:
...
+`for..of` construct:
+
+The bloomrun instance and iterator are all implements iterable protocol,
+patterns can be looped over in a `for..of` construct. Mentioned that, 
+the loop over bloomrun instance will iterate all available patterns in 
+bloomrun instance.
+
+```js
+for (var pattern of instance)
+```
```
Comment (mcollina) [inline README.md]: can you make this about the iterator itelself, and talk about main object at the top (after the constructor)?

File: bloomrun.js (hunk 1 above)
Comment (tz70s) [reply bloomrun.js]: Oops, didn't notice that.

**Comment by tz70s** [comment]
Updated!

**Pull request merged**